lib/stream.js

```diff
@@ -1,11 +1,14 @@
 'use strict';
 
-module.exports = Stream;
+var Stream = require('internal/streams/stream').Stream;
+
+// wrap the old-style stream
+require('internal/streams/legacy')(Stream);
```

Comment:
Yes, but then why split the definition of it in one file, and the legacy part in another?
Either we just move the legacy bits, we move the full definition, or we move all the streams parts.

I'm 👎 in having _both_ `internal/streams/stream` and `internal/streams/legacy`, if we are not moving along readable, writable, etc.